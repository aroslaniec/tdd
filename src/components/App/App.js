import React, { Component } from "react";

import "./App.scss";
import pathToLogoSrc from "../../media/logo.svg";
import Logo from "../Logo";
import MainContent from "../MainContent";

class App extends Component {
  static propTypes = {};
  static defaultProps = {};
  static displayName = "App";

  render() {
    return (
      <div className="app">
        <header className="app__header">
          <Logo pathToSrc={pathToLogoSrc} title="Hello World!" />
          <h1>Let's TDD!</h1>
        </header>
        <main>
          <MainContent />
        </main>
      </div>
    );
  }
}

export default App;
