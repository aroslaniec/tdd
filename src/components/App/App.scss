@import "../../styles/variables";

.app {
}

.app__header {
  background-color: $background-primary;
  height: 6rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  font-size: 1.6rem;
  color: white;
  padding: 0 2.4rem 0 0;

  .logo {
    height: 6rem;
    width: 10rem;
  }

  h1 {
    margin: 0;
    line-height: 1;
  }
}
