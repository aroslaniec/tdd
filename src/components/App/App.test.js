import React from "react";
import { shallow } from "enzyme";

import App from "./App";

// Default test from CRA
/*it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});*/

describe("App", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should have className set accordingly to naming convention", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.hasClass("app")).toBeTruthy();
  });
});
