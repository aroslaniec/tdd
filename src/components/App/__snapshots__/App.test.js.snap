// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="app"
>
  <header
    className="app__header"
  >
    <Logo
      pathToSrc="logo.svg"
      title="Hello World!"
    />
    <h1>
      Let's TDD!
    </h1>
  </header>
  <main>
    <MainContent />
  </main>
</div>
`;
